incubator/hnc/internal/validators/object.go

```diff
@@ -246,7 +246,19 @@ func (o *Object) hasConflict(inst *unstructured.Unstructured) (bool, []string) {
 	// Get a list of conflicting descendants if there's any.
 	for _, desc := range descs {
 		if o.Forest.Get(desc).HasSourceObject(gvk, nm) {
-			conflicts = append(conflicts, desc)
+			// check if the selectors match
+			selector, _ := selectors.GetSelector(inst)
+			treeSelector, _ := selectors.GetTreeSelector(inst)
+			noneSelector, _ := selectors.GetNoneSelector(inst)
```

Comment:
There seems to be a lot of overlap  between the code here and the code in the object reconciler (specifically `shouldPropagateSource`). It's actually the same logic but implemented twice, I think? Can we consolidate that all in one place?

Reply:
Agreed, done